src/Workspaces/Core/Portable/Workspace/Host/EventListener/EventListenerTracker.cs

```diff
@@ -8,65 +8,39 @@
 using System.Linq;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Host
+namespace Microsoft.CodeAnalysis.Host;
+
+/// <summary>
+/// helper type to track whether <see cref="IEventListener"/> has been initialized.
+/// 
+/// currently, this helper only supports services whose lifetime is same as Host (ex, VS)
+/// </summary>
+internal sealed class EventListenerTracker(
+    IEnumerable<Lazy<IEventListener, EventListenerMetadata>> eventListeners, string kind)
 {
-    /// <summary>
-    /// helper type to track whether <see cref="IEventListener"/> has been initialized.
-    /// 
-    /// currently, this helper only supports services whose lifetime is same as Host (ex, VS)
-    /// </summary>
-    /// <typeparam name="TService">TService for <see cref="IEventListener{TService}"/></typeparam>
-    internal sealed class EventListenerTracker<TService>(
-        IEnumerable<Lazy<IEventListener, EventListenerMetadata>> eventListeners, string kind)
-    {
-        /// <summary>
-        /// Workspace kind this event listener is initialized for
-        /// </summary>
-        private readonly HashSet<string> _eventListenerInitialized = [];
-        private readonly ImmutableArray<Lazy<IEventListener, EventListenerMetadata>> _eventListeners = [.. eventListeners.Where(el => el.Metadata.Service == kind)];
-
-        public void EnsureEventListener(Workspace workspace, TService serviceOpt)
```

Comment:
this was never called.  it was the only thing that passed a service.  hence why the service arg on StartListening could be removed.